module.go

```diff
@@ -138,6 +139,7 @@ func NewModuleFromBufferArgs(args NewModuleArgs) (*Module, error) {
 	if optsC == nil {
 		return nil, fmt.Errorf("failed to create bpf_object_open_opts: %w", errno)
 	}
+	optsC.kernel_log_level = args.KernelLogLevel
```

Comment:
Instantiate the C object `kernelLogLevelC` right below `bpfBuffSizeC := C.size_t(len(args.BPFObjBuff))`.

That's to avoid direct assignments to C structs since cgo doesn't work 100% of the cases. With that in mind, you should change the `cgo_bpf_object_open_opts_new()` helper to receive `kernelLogLevelC` and assign it properly.